### PR TITLE
Allow controlling the libdrm dependency

### PR DESCRIPTION
**Pull request by hughsie**
Fixes https://github.com/fwupd/fwupd/issues/7877

Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

**Comment by hughsie** [comment]
@tgurr -- like this?

**Comment by tgurr** [comment]
> @tgurr -- like this?

@hughsie basically yes, thanks!. 

Some questions:
* Is it intended to default to off, e.g. when not passing a `-Dlibdrm=enabled` argument to meson? 
* does `plugin_amdgpu` depend on the option being enabled?
* not exclusive to this: generally more speaking descriptions would be helpful (for packagers) to get an idea for which features the flags are required/recommended as like in this example `libdrm support` doesn't actually add any useful information.

**Comment by hughsie** [comment]
> Is it intended to default to off

It should be default on -- unless you're using `-Dauto_features=disabled`

> does plugin_amdgpu depend on the option being enabled

Well, yes, indirectly -- but we're checking libdrm_amdgpu with IIUC is a superset of libdrm.

> descriptions would be helpful (for packagers

Hmm, I'm not sure the --help output is the best place for that -- but that said anywhere *else* is going to get out of sync really quick.

**Comment by tgurr** [comment]
> > Is it intended to default to off
> 
> It should be default on -- unless you're using `-Dauto_features=disabled`

Thanks for the hint! That was indeed the case.

> > does plugin_amdgpu depend on the option being enabled
> 
> Well, yes, indirectly -- but we're checking libdrm_amdgpu with IIUC is a superset of libdrm.

Thanks. When it's (currently) not required or used for other features/plugins I'd then tie it to the amdgpu plugin option being enabled/disabled in our package.

> > descriptions would be helpful (for packagers
> 
> Hmm, I'm not sure the --help output is the best place for that -- but that said anywhere _else_ is going to get out of sync really quick.

Can't say anything about that just wanted to mention the pain that it is often hard to judge and wrap your head around when not looking at the code and being able to understand it for what features the various options are actually required when not having meaningful descriptions, that problem not being exclusive to fwupd though.

File: meson_options.txt [hunk 1]
```diff
@@ -91,6 +91,10 @@ option('libarchive',
   type: 'feature',
   description: 'libarchive support',
 )
+option('libdrm',
+  type: 'feature',
+  description: 'libdrm support',
+)
```
Comment (superm1) [inline meson_options.txt:97]: IMO we should drop `plugin_amdgpu` meson option if adding this and make _it_ automagic.

Comment (hughsie) [reply meson_options.txt:97]: That's basically what we have, no? or do you mean we should do `s/plugin_amdgpu/libdrm_amdgpu`?

Comment (superm1) [reply meson_options.txt:97]: I think if we add a toplevel `libdrm` meson tri-state feature then we don't need a toplevel `plugin_amdgpu` meson tristate feature anymore.  Just let whether `libdrm` is enabled control whether we get `plugin_amdgpu`

Comment (hughsie) [reply meson_options.txt:97]: Hmm, I'm not sure about this; if you compile libdrm with `-Damdgpu=disabled` then `libdrm.pc` exists, but not `libdrm_amdgpu.pc`

Comment (superm1) [reply meson_options.txt:97]: Ahhhh true.  OK.

**Review by superm1 — APPROVED** [review]

**Pull request merged**